Add method to save Fiji script parameters, and add example to opening an image

rohangirishrao opened the pull request:
Safe way of parsing Fiji script params by ignoring the passwords field automatically.

rohangirishrao committed:
Add example to docstring for clarity

rohangirishrao committed:
Add method to save Fiji script parameters to file

Calls on scijava.script.ScriptModule through Python globals to access only the user-given parameters and not other variables at run-time.

rohangirishrao committed:
Add keys to skip logging

rohangirishrao committed:
Update method to save Fiji script parameters

The method now ignores password in a clean way, and skips some other
non-essential runtime keys.

rohangirishrao committed:
Run black formatter

rohangirishrao committed:
Remove redundant ScriptModule import

rohangirishrao committed:
Remove redundant comments

imagejan commented on src/imcflibs/imagej/misc.py:
```diff
@@ -701,3 +701,58 @@ def run_imarisconvert(file_path):
+            # Skip if parameter is declared with style="password"
+            style = item.getWidgetStyle()
+            if style is not None and style.lower() == "password":
+                continue
```
You could use [`WidgetStyle.isStyle(ModuleItem<?> item, String target)`](https://javadoc.scijava.org/SciJava/index.html?org/scijava/widget/WidgetStyle.html) here:


```suggestion
            # Skip if parameter is declared with style="password"
            if WidgetStyle.isStyle(item, TextWidget.PASSWORD_STYLE):
                continue
```

Note also the use of public constants instead of hard-coded strings here 😉

ehrenfeu replied:
I like @imagejan's suggestion. @rohangirishrao  could you look into it?

ehrenfeu commented on src/imcflibs/imagej/omerotools.py:
For future PR's, please try to not have completely unrelated stuff in one PR as this is more difficult to review (sure, not in the case of a docstring change, but still) and will potentially block stuff from being merged.

rohangirishrao replied:
Copy that!

rohangirishrao committed:
Change from string literal to public constant

ehrenfeu commented:
Add to changelog #126

imagejan commented:
From the GitHub Actions checks:

```
src/imcflibs/imagej/misc.py:19: in <module>
    from org.scijava.widget import WidgetStyle
E   ModuleNotFoundError: No module named 'org'
```

Looks like you need to update `imcf-fiji-mocks` now (and bump its version here) to provide the necessary `org.scijava` classes.

rohangirishrao committed:
Change mocks version to 0.12.0

ehrenfeu commented:
Yep: https://github.com/imcf/imcf-fiji-mocks/releases/tag/v0.12.0 😁 

Thanks for spotting!

> From the GitHub Actions checks:
> 
> ```
> src/imcflibs/imagej/misc.py:19: in <module>
>     from org.scijava.widget import WidgetStyle
> E   ModuleNotFoundError: No module named 'org'
> ```
> 
> Looks like you need to update `imcf-fiji-mocks` now (and bump its version here) to provide the necessary `org.scijava` classes.

ehrenfeu committed:
Update poetry.lock for mocks v0.12.0

ehrenfeu commented:
🕵🏼 Need to investigate why `Pytest-Python2` is failing, probably the [culprit is in src/imcflibs/imagej/misc.py][1]

```
____________ ERROR collecting tests/bdv/test_define_dataset_auto.py ____________
venv.py2/lib/python2.7/site-packages/_pytest/python.py:507: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
venv.py2/lib/python2.7/site-packages/py/_path/local.py:704: in pyimport
    __import__(modname)
venv.py2/lib/python2.7/site-packages/_pytest/assertion/rewrite.py:304: in load_module
    exec(co, mod.__dict__)
tests/bdv/test_define_dataset_auto.py:6: in <module>
    from imcflibs.imagej import bdv
src/imcflibs/imagej/__init__.py:9: in <module>
    from . import misc
E     File "/home/runner/work/python-imcflibs/python-imcflibs/src/imcflibs/imagej/misc.py", line 743
E   SyntaxError: Non-ASCII character '\xe2' in file /home/runner/work/python-imcflibs/python-imcflibs/src/imcflibs/imagej/misc.py on line 743, but no encoding declared; see http://python.org/dev/peps/pep-0263/ for details
```

[1]: https://github.com/imcf/python-imcflibs/actions/runs/21033926547/job/60476621832?pr=111#step:8:180

ehrenfeu committed:
Replace UTF-8 dash with an ASCII minus

Using UTF-8 is still shaky in the Jython 2.7 environment, so if there is
no strong need for it, simply stick to ASCII.

ehrenfeu committed:
Use IJ.log() instead of print()

Also see #45

ehrenfeu committed:
Minor syntax / rendering fix

ehrenfeu committed:
Update mocks dependency to version to 0.13.0

ehrenfeu committed:
Add instructions for working around poetry's stubborn cache

ehrenfeu commented on src/imcflibs/imagej/misc.py:
```diff
@@ -701,3 +704,57 @@ def run_imarisconvert(file_path):
+                val = inputs.get(key)
+                f.write("%s: %s\n" % (key, str(val)))
+
+    print("Saved script parameters to: %s" % out_path)
```
cf. #45

ehrenfeu reviewed (approved):
Ready to be merged.
